code/FulpstationCode/body_camera_files/body_camera_procs.dm

```diff
@@ -0,0 +1,101 @@
+#define SEC_BODY_CAM_SOUND list('sound/machines/beep.ogg')
+
+/obj/item/clothing/under/rank/security/Initialize()
+	. = ..()
+	builtInCamera = new (src)
+	builtInCamera.network = list("sec_bodycameras")
+	builtInCamera.internal_light = FALSE
+
+	RegisterSignal(src, COMSIG_ITEM_EQUIPPED, .proc/auto_register_bodycam)
+
+
+/obj/item/clothing/under/rank/security/proc/auto_register_bodycam(datum/source, mob/user, slot)
+	if(!builtInCamera)
+		return
+	if(slot != SLOT_W_UNIFORM)
+		return
+	if(!user)
+		if(ismob(loc))
+			user = loc
+		else
+			return
+	var/obj/item/card/id/I = user.get_idcard(TRUE)
+	if(!istype(I))
+		return
+	if(check_access(I))
+		var/id_name = I.registered_name
+		builtInCamera.c_tag = "*Body Camera: [I.assignment] [id_name]"
+		camera_sound()
+		to_chat(user, "<span class='notice'>Security uniform body camera automatically registered to [id_name]</span>")
+
+/obj/item/clothing/under/rank/security/attackby(obj/item/W, mob/user, params)
+	. = ..()
+	if(!builtInCamera)
+		to_chat(user, "<span class='warning'>No body camera detected.</span>")
+		return
+
+	var/obj/item/card/id/I
+	if (istype(W, /obj/item/card/id))
+		I = W
+	else if (istype(W, /obj/item/pda))
+		var/obj/item/pda/P = W
+		I = P.id
+
+	if(!I)
+		to_chat(user, "<span class='warning'>No ID detected for body camera registration.</span>")
+		return
+
+	if(check_access(I))
+		var/id_name = I.registered_name
+		builtInCamera.c_tag = "*Body Camera: [I.assignment] [id_name]"
+		camera_sound()
+		to_chat(user, "<span class='notice'>Security uniform body camera manually registered with ID to [id_name]</span>")
+	else
+		to_chat(user, "<span class='warning'>ID is not authorized for registration with this uniform's body camera.</span>")
+
+/obj/item/clothing/under/rank/security/verb/toggle_camera()
+	set name = "Toggle Body Camera"
+	set category = "Object"
+	set src in usr
+	var/mob/M = usr
+	if (istype(M, /mob/dead/))
+		return
+	if (!can_use(M))
+		return
+	camera_toggle(usr)
+
+
+/obj/item/clothing/under/rank/security/proc/camera_toggle()
+	var/message = "<span class='notice'>There's no camera!</span>"
+
+	if(!builtInCamera)
+

```

Comment:
Is there supposed to be anything checked under !builtInCamera?
You've got an empty if-statement
If the intention is to catch the !builtInCamera and then throw it, to where you skip that section of checks and proceed to the if(ismob(loc)) then why not do the reverse and check for builtInCamera?, thus reducing the amount of code needed

Reply:
I meant to delete that. Fixed.